lib/src/over_react_test/common_component_util.dart

```diff
@@ -406,25 +409,67 @@ void testRequiredProps(BuilderOnlyUiFactory factory, dynamic childrenFactory())
+    if (!useComponent2Tests) {
+      requiredProps.forEach((String propKey) {
+        final reactComponentFactory = factory().componentFactory as ReactDartComponentFactoryProxy; // ignore: avoid_as
 
-      // Props that are defined in the default props map will never not be set.
-      if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+        // Props that are defined in the default props map will never not be set.
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+          var badRenderer = () => render((factory()
+            ..remove(propKey)
+          )(childrenFactory()));
+
+          expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is not set');
+        }
+
+        var propsToAdd = {propKey: null};
         var badRenderer = () => render((factory()
-          ..remove(propKey)
+          ..addAll(propsToAdd)
         )(childrenFactory()));
 
-        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is not set');
-      }
+        expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is set to null');
+      });
+    } else {
+      PropTypes.resetWarningCache();
+
+      List<String> consoleErrors = [];
+      JsFunction originalConsoleError = context['console']['error'];
+      context['console']['error'] = new JsFunction.withThis((self, [message, arg1, arg2, arg3,  arg4, arg5]) {
+        consoleErrors.add(message);
+        originalConsoleError.apply([message], thisArg: self);
+      });
 
-      var propsToAdd = {propKey: null};
-      var badRenderer = () => render((factory()
-        ..addAll(propsToAdd)
-      )(childrenFactory()));
+      final reactComponentFactory = factory().componentFactory as
+          ReactDartComponentFactoryProxy2; // ignore: avoid_as
 
-      expect(badRenderer, throwsPropError_Required(propKey, keyToErrorMessage[propKey]), reason: '$propKey is set to null');
-    });
+      requiredProps.forEach((String propKey) {
+        if (!reactComponentFactory.defaultProps.containsKey(propKey)) {
+
+          mount((factory()
+            ..remove(propKey)
+          )(childrenFactory()));
+
+          expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
+          expect(consoleErrors, [contains(keyToErrorMessage[propKey])],
+              reason: '$propKey is not set');
+        }
+
+        context['console']['error'] = originalConsoleError;
+
+        var propsToAdd = {propKey: null};
+
+        mount((factory()
+          ..addAll(propsToAdd)
+        )(childrenFactory()));
+
+        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
+        expect(consoleErrors, [contains(keyToErrorMessage[propKey])],
+            reason: '$propKey is not set');
+
+
+        context['console']['error'] = originalConsoleError;
```

Comment:
Restoring this at the bottom at the test comes with a risk: if something in the test above throws (either due to an unexpected exception or a `expect` failure), it could result in this not getting run, leaving the console in a bad state for subsequent tests, which could cause confusing errors and failures that might not have otherwise occurred.

Could you do it in an `addTearDown`, which always gets run even if the test fails? 
```diff
 context['console']['error'] = new JsFunction.withThis( ... );
+addTearDown(() => context['console']['error'] = originalConsoleError);
```
A `tearDown` block would work as well, but `addTearDown` is nice when it's specific to just a single test and you don't want to create a whole new group just to add tear-down logic.

Reply:
That's awesome, thank you, Greg!